src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,62 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+type Props = {
+  todoSelector: string;
+  onChangeTodoSelector: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+  todoFilter: string;
+  onChangeTodoFilter: (
+    event:
+    | React.ChangeEvent<HTMLInputElement>
+    | React.MouseEvent<HTMLButtonElement>,
+  ) => void;
+};
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = ({
+  todoSelector,
+  onChangeTodoSelector,
+  todoFilter,
+  onChangeTodoFilter,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={todoSelector}
+            onChange={onChangeTodoSelector}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={todoFilter}
+          onChange={onChangeTodoFilter}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          {todoFilter !== '' && (
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={onChangeTodoFilter}
```

Comment:
Better crate clear handler where you will set empty query
Refactor `handleTodoFiltering ` after changes
```
const onClearQuery = () => {
  onChangeTodoFilter('')
}
```